api/src/main/java/org/openmrs/BaseOpenmrsMetadata.java

```diff
@@ -22,29 +25,40 @@
  * @since 1.5
  * @see OpenmrsMetadata
  */
+@MappedSuperclass
 public abstract class BaseOpenmrsMetadata extends BaseOpenmrsObject implements OpenmrsMetadata {
 	
 	//***** Properties *****
+	@Column(name = "name", nullable = false, length = 255)
 	@Field
 	private String name;
 	
+	@Column(name = "description", length = 255)
```

Comment:
This is wrong. You should not have overridden any existing classes. In fact annotations are ignored for entities mapped in xml. Reading Hibernate docs: "To ease the migration process from hbm files to annotations, the configuration mechanism detects the mapping duplication between annotations and hbm files. HBM files are then prioritized over annotated metadata on a class to class basis. You can change the priority using hibernate.mapping.precedence property. The default is hbm, class, changing it to class, hbm will prioritize the annotated classes over hbm files when a conflict occurs." https://docs.jboss.org/hibernate/stable/annotations/reference/en/html/ch01.html
So please revert changes that you made after Wyclif's comment.

The correct way to override any field is by using @AttributeOverwrite, but you must not use it for this issue. The issue is only about introducing annotations to Base.. classes. http://stackoverflow.com/questions/5257921/hibernate-how-override-an-attribute-from-mapped-super-class